airbyte-commons/src/main/java/io/airbyte/commons/json/JsonSchemas.java

```diff
@@ -8,15 +8,48 @@
 import com.fasterxml.jackson.databind.node.ObjectNode;
 import io.airbyte.commons.io.IOs;
 import io.airbyte.commons.resources.MoreResources;
+import io.airbyte.commons.util.MoreIterators;
 import java.io.IOException;
 import java.nio.file.Files;
 import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Iterator;
 import java.util.List;
+import java.util.Map.Entry;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.BiConsumer;
+import java.util.function.BiFunction;
+import java.util.function.Predicate;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
+// todo handle capitalization
+// todo we don't hande not. fail explicitly if we enoucnter it
 public class JsonSchemas {
 
+  private static final String JSON_SCHEMA_TYPE_KEY = "type";
+  private static final String JSON_SCHEMA_PROPERTIES_KEY = "properties";
+  private static final String JSON_SCHEMA_ITEMS_KEY = "items";
+
+  // all JSONSchema types.
+  private static final String ARRAY_TYPE = "array";
+  private static final String OBJECT_TYPE = "object";
+  private static final String STRING_TYPE = "string";
+  private static final String NUMBER_TYPE = "number";
+  private static final String BOOLEAN_TYPE = "boolean";
+  private static final String NULL_TYPE = "null";
+  private static final String ONE_OF_TYPE = "oneOf";
+  private static final String ALL_OF_TYPE = "allOf";
+  private static final String ANY_OF_TYPE = "anyOf";
+
+  private static final String ARRAY_JSON_PATH = "[]";
+
+  private static final Set<String> COMPOSITE_KEYWORDS = Set.of(ONE_OF_TYPE, ALL_OF_TYPE, ANY_OF_TYPE);
```

Comment:
We are missing nested here.

Reply:
i'm not that familiar with that keyword. can you point it out to me in the jsonschema docs? i can't find it.

Reply:
My bad, I think that it was a keyword but it is just name of the key in the specs of the test.